activemodel/lib/active_model/attribute_methods_builder.rb

```diff
@@ -0,0 +1,225 @@
+# frozen_string_literal: true
+
+require "concurrent/map"
+
+module ActiveModel
+  class AttributeMethodsBuilder < Module # :nodoc:
+    NAME_COMPILABLE_REGEXP = /\A[a-zA-Z_]\w*[!?=]?\z/
+    CALL_COMPILABLE_REGEXP = /\A[a-zA-Z_]\w*[!?]?\z/
+
+    attr_accessor :matchers
+
+    def initialize
+      @matchers = [AttributeMethodMatcher.new]
+      @method_names = Set.new
+    end
+
+    def included(model_class)
+      # Strictly-speaking this is not necessary, since AM::AttributeMethods
+      # includes the module, and AR::AttributeMethods includes
+      # AM::AttributeMethods. However, since this class depends on methods in
+      # it, we should ensure that it is included into any module that includes
+      # instances of the builder.
+      model_class.include AttributeMissingMethods
```

Comment:
What about including it into the module itself, in `initialize`?

I'm also unclear about why this is in both places: the above comment explains why it should be here, and not only in AM::AttributeMethods... but why can't it only be here? I'm probably missing something obvious about the relationship.

Reply:
It's included explicitly in AM::AttributeMethods because of the dependency on when these methods are included into the heirarchy tree. If you don't explicitly include this in AM::AttributeMethods, then these methods end up getting included higher up and stuff breaks in AR (but not in AM).

Reply:
About including it in `initialize`, this seems to work as well (just testing now locally). I think I had it in `includes` for a reason that is not relevant anymore. I can move it.

Reply:
I just moved it in 120fc31005694e0e1e60a6ee8a345cd2913d073b